Frends.ServiceBus/ServiceBus.cs

```diff
@@ -341,6 +348,109 @@ private static Encoding GetEncodingFromContentType(string contentTypeString, Enc
                 }
             }
         }
+
+        static async public Task<ReadResult> SendAndWaitForResponse(SendAndWaitForResponseInput input, SendAndWaitForResponseOptions options, CancellationToken token = new CancellationToken())
+        {
+            // Prepare queues
+            if (options.CreateQueues)
+            {
+                await EnsureQueueExists(input.Queue, input.ConnectionString, false);
+                await EnsureQueueExists(input.ReplyToQueue, input.ConnectionString, true);
+            }
+
+            QueueClient requestClient = MessagingFactory.CreateFromConnectionString(input.ConnectionString).CreateQueueClient(input.Queue);
```

Comment:
We talked about caching all of that already and I can honestly say that I do not have that much experience with SB API, so it would probably be best if Erkka or some other SB guru would look at this and do it the correct way.

As for topics and stuff - I understood (maybe wrongly) that subscriptions are there for a slightly different use case. Sessions allow request/response pattern in SB, but subscriptions are not directly targeting this scenario. They are just for looking into messages with specific attributes. You can probably imitate session functionality with those.